content/chainguard/migration-guides/tutorial/index.md

```diff
@@ -0,0 +1,732 @@
+Python](https://hub.docker.com/_/python) images makes sense, but can still be a source of confusion.
+Fixing this can be as easy as changing the entrypoint, but let's take a look at the script first.
+This is the file `cmd.sh` in the identidock directory:
+
+```bash
+#!/bin/bash
+set -e
+
+if [ "$ENV" = 'DEV' ]; then
+   echo "Running Development Server"
+   exec python "/app/identidock.py"
+elif [ "$ENV" = 'UNIT' ]; then
+    echo "Running Unit Tests"
+    exec python "tests.py"
+else
+    echo "Running Production Server"
+    exec uwsgi --http 0.0.0.0:9090 --wsgi-file /app/identidock.py \
+               --callable app --stats 0.0.0.0:9191
+fi
+```
+
+This script decides how to run the application depending on how the `ENV` environment variable is
+set. The idea here is to allow us to use the same image in development, testing and production. This
+approach is no longer recommended as it leads to development tooling being present in the production
+environment. Even though the development tooling isn't run in production, it is still bloating the
+image and is potentially exploitable by attackers. Therefore, we will use a different approach and
+break the Dockerfile into separate development and production images.
+
+Let’s skip to the final Dockerfile for our image and walk through the changes made. These changes
+address multiple issues, beyond just having multiple images, and are based on the [Chainguard
+Academy guide to Python
+images](https://edu.chainguard.dev/chainguard/chainguard-images/getting-started/python/).
+
+Replace the Dockerfile with this one (this is also available on the "main" branch):
+
+```Dockerfile
+FROM cgr.dev/chainguard/python:latest-dev as dev
+
+ENV LANG=C.UTF-8
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PYTHONUNBUFFERED=1
+ENV PATH="/app/venv/bin:$PATH"
+
+WORKDIR /app
+RUN python -m venv /app/venv
+COPY requirements.txt ./
+RUN pip install --no-cache-dir -r requirements.txt
+COPY app /app
+
+EXPOSE 5000
+
+ENTRYPOINT ["python"]
+CMD ["identidock.py"]
+
+FROM cgr.dev/chainguard/python
+
+WORKDIR /app
+ENV PYTHONUNBUFFERED=1
+ENV PATH="/app/venv/bin:$PATH"
+
+COPY app/identidock.py ./
+COPY --from=dev /app/venv /app/venv
+
+EXPOSE 9090
+ENTRYPOINT [ "gunicorn", "-b", "0.0.0.0:9090", "identidock:app" ]
+```
+
+And add the file `requirements.txt` to the current directory (`identidock`) with the following contents:
+
+```
+Flask==3.0.2
+requests==2.31.0
+redis==5.0.3
+gunicorn==21.2.0
+```
+
+The first thing to notice is that we have a multistage build now. If you want the development image
+rather than the production one, you can specify it during docker build:
+
+```bash
+docker build --target dev -t identidock:dev .
+```
+
+Otherwise, you will get the production image only.
+
+There are several more environment variables defined. These prevent the creation of Python bytecode
+and buffering of output. The reasons why you want these are normally set are explained in detail in
```

Comment:
"The reasons why you want these are normally set are explained" reads funny to me

Reply:
rewritten